scripts/save

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env node
+
+const fs = require('fs')
+const path = require('path')
+const Web3 = require('web3')
+const APM = require('@aragon/apm')
+const execa = require('execa')
+const yaml = require('js-yaml')
+const { promisify } = require('util')
+const mkdirp = promisify(require('mkdirp'))
+const dotprop = require('dot-prop')
+
+const defaultAPM = require('./helpers/default-apm')
+const ENVIRONMENTS = require('./helpers/environments')
+
+const NEW_VERSION_LOG_ABI = [{ type: 'uint256', name: 'versionId' }, { type: 'uint16[3]', name: 'semanticVersion'}]
+
+const getArchiveSavePath = (env, repo, version) =>
+ `environments/${env}/${repo}/${repo}@${version}`
+
+const ipfsGetCommand = (hash, output) =>
+  `get -a -C -o ${output} ${hash}`
+
+const getAndSaveIPFSFile = async (hash, output) => {
+  const { stdout } = await execa('ipfs', ipfsGetCommand(hash, output).split(' '))
+
+  console.log(stdout)
+}
+
+const getPublishedVersion = async (txHash, repoName, { node, ens }) => {
+  const web3 = new Web3(node)
+  const apm = APM(web3, { ensRegistryAddress: ens })
+
+  const receipt = await web3.eth.getTransactionReceipt(txHash)
+
+  const repoAddress = await apm.ensResolve(repoName)
+
+  // Only supports one log per transaction
+  const log = receipt.logs.find(
+    ({ address }) => address.toLowerCase() === repoAddress.toLowerCase()
+  )
+
+  const { semanticVersion } = web3.eth.abi.decodeLog(NEW_VERSION_LOG_ABI, log.data, log.topics)
+
+  return apm.getVersion(repoName, semanticVersion, 2000)
+}
+
+const getTransactionTimestamp = async (txHash, { node }) => {
+  const web3 = new Web3(node)
+  const { blockNumber } = await web3.eth.getTransaction(txHash)
+
+  if (!blockNumber) {
+    throw new Error(`Transaction ${txHash} hasn't been included in a block yet`)
+  }
+
+  const { timestamp } = await web3.eth.getBlock(blockNumber)
+
+  return timestamp
+}
+
+const saveDeploymentEntry = (environmentName, repoName, version, entry) => {
+  const filename = path.join('environments', environmentName, 'deploys.yml')
```

Comment:
Might want to preface this with `__dirname` and `../environments` similar to above.